_posts/2018-03-01-spatial-modeling-mlr.markdown

```diff
@@ -0,0 +1,318 @@
+---
+author: Patrick Schratz
+biblio-style: apalike
+bibliography: ../bibs/spatial\_modeling\_`mlr`.bib
+categories: r
+comments: True
+date: 01 March, 2018
+layout: post
+---
+link-citations: True
+title: A practical guide to performance estimation of spatially tuned
+  machine-learning models for spatial data using `mlr`
+
+<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.0/MathJax.js?config=TeX-AMS-MML_HTMLorMML" type="text/javascript"></script>
+* TOC 
+{:toc}
+
+\[[view raw
+Rmd](https://raw.githubusercontent.com//r-spatial/r-spatial.org/gh-pages/_rmd/2018-03-01-spatial-modeling-mlr.Rmd)\]
+
+Introduction
+============
+
+Recently we started to integrate the spatial partitioning methods of
+[sperrorest](https://mlr-org.github.io/mlr-tutorial/devel/html/handling_of_spatial_data/index.html)
+into [mlr](http://mlr-org.github.io). `mlr` is a unified interface to
+conduct all kind of modeling (similar to `caret`), currently supporting
+more than 100 modeling packages. In comparison to sperrorest it
+excelaertes by providing the option to easily tune hyperparameters. We
+started with the most common spatial partitioning approach which uses
+k-means clustering (see Brenning ([2012](#ref-sperrorest))). In this
+blog post I will give a practical guide on how to perform a nested
+spatial cross-validation in `mlr`. I will use a SVM as the algorithm
+because it is a widely know and used algorithm that always needs to be
+tuned to achieve good performance. Although the tuning effect of models
+with meaningful default hyperparameter values such as Random Forest is
+not high, one should always conduct a hyperparameter tuning as the
+specific effect is not known a priori.
```

Comment:
Is this general wisdom, or yours? In case of the first, maybe add a reference?

Reply:
Two-fold:

> Although the tuning effect of models with meaningful default hyperparameter values such as Random Forest is not high

This is logic based; meaningful hyperpar = optimized by algorithm author to perform somewhat good in most cases. Therefore tuning effect rather small compared to arbitrary hyperpars (e.g. SVM),

> one should always conduct a hyperparameter tuning as the specific effect is not known a priori

This is a combination of logic and personal wisdom. 

I'm unsure if there is a need for a citation here as the statements are mainly logic based rather than relying on the findings of a study. However, I could add this thought for clarification?